// mobile/src/features/auth/LoginScreen.styles.js
import { StyleSheet, Dimensions, Platform } from 'react-native';

const { width: SCREEN_WIDTH } = Dimensions.get('window');
const scale = (size) => (SCREEN_WIDTH / 408) * size;

export default StyleSheet.create({
  container: {
    flex: 1,
  },
  header: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: scale(40),
  },
  logo: {
    width: scale(72),
    height: scale(72),
    marginBottom: 12,
    tintColor: '#fff',
  },
  title: {
    color: '#fff',
    fontSize: scale(26),
    fontWeight: '900',
    letterSpacing: 0.5,
  },
  subtitle: {
    color: '#ccc',
    fontSize: scale(14),
    marginTop: 4,
  },
  formContainer: {
    backgroundColor: '#F7F7F7',
    borderTopLeftRadius: 40,
    borderTopRightRadius: 40,
    paddingHorizontal: scale(30),
    paddingTop: scale(40),
    paddingBottom: Platform.OS === 'ios' ? scale(45) : scale(30),
    width: '100%',
    elevation: 20,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: -10 },
    shadowOpacity: 0.1,
    shadowRadius: 15,
  },
  group: {
    marginBottom: scale(20),
  },
  labelRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 8,
  },
  label: {
    fontWeight: '700',
    fontSize: scale(13),
    color: '#444',
  },
  errorText: {
    color: '#DC2626',
    fontSize: scale(11),
    fontWeight: '700',
  },
  inputGroup: {
    flexDirection: 'row',
    alignItems: 'center',
    borderWidth: 1.5,
    borderColor: '#E8E8E8',
    borderRadius: 16,
    paddingHorizontal: 16,
    height: scale(58),
    backgroundColor: '#FFF',
  },
  errorBorder: {
    borderColor: '#FCA5A5',
    backgroundColor: '#FFF8F8',
  },
  input: {
    flex: 1,
    paddingHorizontal: 12,
    fontSize: scale(15),
    color: '#111',
    fontWeight: '500',
  },
  controls: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginVertical: scale(12),
  },
  checkbox: {
    width: 20,
    height: 20,
    borderRadius: 6,
    borderWidth: 2,
    borderColor: '#DDD',
    marginRight: 10,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#FFF',
  },
  checkboxActive: {
    backgroundColor: '#000',
    borderColor: '#000',
  },
  remember: {
    fontSize: scale(13),
    color: '#666',
    fontWeight: '600',
  },
  forgot: {
    fontSize: scale(13),
    color: '#000',
    fontWeight: '800',
  },
  generalErrorBox: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#FEF2F2',
    padding: 14,
    borderRadius: 14,
    marginBottom: 18,
    borderWidth: 1,
    borderColor: '#FEE2E2',
  },
  generalErrorText: {
    color: '#B91C1C',
    fontSize: scale(12),
    fontWeight: '700',
    marginLeft: 10,
    flex: 1,
    lineHeight: 18,
  },
  loginBtn: {
    backgroundColor: '#000',
    height: scale(58),
    borderRadius: 16,
    alignItems: 'center',
    justifyContent: 'center',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.15,
    shadowRadius: 10,
    elevation: 4,
  },
  loginBtnDisabled: {
    backgroundColor: '#333',
    opacity: 0.8,
  },
  loginText: {
    color: '#fff',
    fontWeight: '800',
    fontSize: scale(16),
    letterSpacing: 0.5,
  },
  divider: {
    textAlign: 'center',
    color: '#BBB',
    marginVertical: scale(25),
    fontSize: scale(12),
    fontWeight: '600',
    textTransform: 'uppercase',
    letterSpacing: 1,
  },
  socials: {
    flexDirection: 'row',
    justifyContent: 'center',
    gap: scale(18),
  },
  socialBtn: {
    width: scale(54),
    height: scale(54),
    borderRadius: 18,
    backgroundColor: '#FFF',
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 1,
    borderColor: '#EEE',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.05,
    shadowRadius: 4,
    elevation: 1,
  },
  socialIcon: {
    width: 26,
    height: 26,
  },
  footer: {
    marginTop: scale(30),
    textAlign: 'center',
    color: '#888',
    fontSize: scale(14),
    fontWeight: '600',
  },
  signUpLink: {
    color: '#000',
    fontWeight: '900',
    textDecorationLine: 'underline',
  }
});
