// mobile/src/features/customer/DashboardScreen.styles.js
import { StyleSheet, Dimensions, Platform, StatusBar } from 'react-native';

const { width: SCREEN_WIDTH, height: SCREEN_HEIGHT } = Dimensions.get('window');

const guidelineBaseWidth = 408;
const guidelineBaseHeight = 906;

const scale = (size) => (SCREEN_WIDTH / guidelineBaseWidth) * size;
const verticalScale = (size) => (SCREEN_HEIGHT / guidelineBaseHeight) * size;
const moderateScale = (size, factor = 0.5) => size + (scale(size) - size) * factor;

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: '#F8F9FA' },
  header: {
    backgroundColor: '#111',
    paddingHorizontal: scale(20),
    paddingBottom: verticalScale(25),
    paddingTop: Platform.OS === 'ios' ? verticalScale(50) : (StatusBar.currentHeight || 0) + verticalScale(10),
    borderBottomLeftRadius: scale(30),
    borderBottomRightRadius: scale(30),
  },
  headerTop: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center' },
  greeting: { color: '#fff', fontSize: moderateScale(22), fontWeight: 'bold' },
  notificationWrapper: { backgroundColor: '#fff', padding: scale(10), borderRadius: scale(12) },
  notificationDot: { position: 'absolute', top: scale(10), right: scale(12), width: scale(8), height: scale(8), backgroundColor: '#FF3B30', borderRadius: scale(4), borderWidth: 1.5, borderColor: '#fff' },
  searchBar: { backgroundColor: '#fff', borderRadius: scale(15), marginTop: verticalScale(20), flexDirection: 'row', alignItems: 'center', paddingHorizontal: scale(15), height: verticalScale(50) },
  searchInput: { marginLeft: scale(10), flex: 1, fontSize: moderateScale(14), color: '#000' },
  body: { padding: scale(20), paddingBottom: verticalScale(120) },
  categoryChip: { backgroundColor: '#fff', paddingHorizontal: scale(18), paddingVertical: verticalScale(10), borderRadius: scale(12), marginRight: scale(10), borderWidth: 1, borderColor: '#eee' },
  categoryText: { fontSize: moderateScale(13), fontWeight: '600', color: '#555' },
  activeServiceWrapper: { backgroundColor: '#FFF', borderRadius: 24, padding: 20, marginBottom: 25, borderWidth: 1, borderColor: '#F0F0F0', elevation: 5 },
  serviceHeader: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', marginBottom: 15 },
  liveIndicator: { flexDirection: 'row', alignItems: 'center', backgroundColor: '#F0F0F0', paddingHorizontal: 10, paddingVertical: 5, borderRadius: 20 },
  liveDot: { width: 6, height: 6, borderRadius: 3, backgroundColor: '#111', marginRight: 6 },
  liveText: { fontSize: 10, fontWeight: '800', color: '#111' },
  ticketId: { fontSize: 12, color: '#AAA', fontWeight: '600' },
  serviceMain: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', marginBottom: 20 },
  mainServiceTitle: { fontSize: 20, fontWeight: '800', color: '#111' },
  techStatus: { fontSize: 13, color: '#666', marginTop: 4 },
  activeAvatar: { width: 50, height: 50, borderRadius: 15 },
  progressContainer: { marginBottom: 20 },
  progressBarBg: { height: 6, backgroundColor: '#F0F0F0', borderRadius: 3, overflow: 'hidden', marginBottom: 8 },
  progressBarFill: { height: '100%', backgroundColor: '#111', borderRadius: 3 },
  etaLabel: { fontSize: 12, fontWeight: '600', color: '#888' },
  trackButton: { backgroundColor: '#111', flexDirection: 'row', justifyContent: 'center', alignItems: 'center', paddingVertical: 14, borderRadius: 15 },
  trackButtonText: { color: '#FFF', fontWeight: 'bold', fontSize: 14, marginRight: 8 },
  quickActions: { flexDirection: 'row', justifyContent: 'space-between', marginBottom: verticalScale(25) },
  actionItem: { alignItems: 'center' },
  actionCircle: { width: scale(55), height: scale(55), backgroundColor: '#fff', borderRadius: scale(18), justifyContent: 'center', alignItems: 'center', marginBottom: verticalScale(8), elevation: 2, borderWidth: 1, borderColor: '#f0f0f0' },
  actionText: { fontSize: moderateScale(12), fontWeight: '600', color: '#111' },
  sectionHeader: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', marginBottom: verticalScale(15) },
  sectionTitle: { fontSize: moderateScale(18), fontWeight: 'bold', color: '#111' },
  seeAll: { color: '#888', fontSize: moderateScale(13), fontWeight: '600' },
  gridItem: { width: SCREEN_WIDTH * 0.45, height: verticalScale(140), backgroundColor: '#000', borderRadius: scale(20), marginRight: scale(15), overflow: 'hidden' },
  gridImage: { width: '100%', height: '100%', position: 'absolute', opacity: 0.7 },
  gridOverlay: { flex: 1, backgroundColor: 'rgba(0,0,0,0.1)', justifyContent: 'flex-end', padding: scale(15) },
  gridItemText: { color: '#fff', fontWeight: 'bold', fontSize: moderateScale(14) },
  onlineTechCard: { backgroundColor: '#FFF', width: scale(160), borderRadius: scale(25), padding: scale(16), marginRight: scale(15), borderWidth: 1, borderColor: '#F0F0F0', elevation: 3 },
  techImageContainer: { marginBottom: verticalScale(12), alignSelf: 'center' },
  onlineTechImage: { width: scale(65), height: scale(65), borderRadius: scale(20), backgroundColor: '#F8F9FA' },
  onlineIndicatorDot: { position: 'absolute', bottom: 2, right: 2, width: scale(14), height: scale(14), borderRadius: scale(7), backgroundColor: '#4CAF50', borderWidth: 2, borderColor: '#FFF' },
  techInfoCenter: { alignItems: 'center', marginBottom: 10 },
  onlineNameText: { fontSize: moderateScale(14), fontWeight: '800', color: '#111', marginBottom: 2 },
  specialtyRow: { flexDirection: 'row', alignItems: 'center' },
  onlineSubText: { fontSize: moderateScale(11), color: '#888', marginLeft: 4, fontWeight: '500' },
  distanceRow: { flexDirection: 'row', justifyContent: 'center', alignItems: 'center', marginBottom: verticalScale(12) },
  statusBadge: { flexDirection: 'row', alignItems: 'center' },
  statusDotGreen: { width: 6, height: 6, borderRadius: 3, backgroundColor: '#4CAF50', marginRight: 4 },
  onlineStatusText: { fontSize: moderateScale(10), color: '#4CAF50', fontWeight: '700' },
  locationBadge: { flexDirection: 'row', alignItems: 'center' },
  onlineDistanceText: { fontSize: moderateScale(10), color: '#888', marginLeft: 3, fontWeight: '600' },
  dotSeparator: { marginHorizontal: 5, color: '#DDD' },
  requestActionBtn: { backgroundColor: '#111', width: '100%', paddingVertical: verticalScale(10), borderRadius: scale(12), alignItems: 'center' },
  requestActionText: { color: '#FFF', fontSize: moderateScale(12), fontWeight: 'bold' },
  historyContainer: { backgroundColor: '#FFF', borderRadius: scale(20), padding: scale(5), marginBottom: verticalScale(10), borderWidth: 1, borderColor: '#EEE' },
  historyItem: { flexDirection: 'row', alignItems: 'center', padding: scale(15), borderBottomWidth: 1, borderBottomColor: '#F5F5F5' },
  historyIconBox: { width: scale(45), height: scale(45), borderRadius: scale(12), justifyContent: 'center', alignItems: 'center' },
  historyTitle: { fontSize: moderateScale(14), fontWeight: 'bold', color: '#111' },
  statusBadgeContainer: { marginTop: 4 },
  statusTagText: { fontSize: moderateScale(10), fontWeight: '800', textTransform: 'uppercase' },
  emptyHistory: { padding: scale(40), alignItems: 'center', backgroundColor: '#FFF', borderRadius: scale(20), borderStyle: 'dashed', borderWidth: 1, borderColor: '#DDD' },
  emptyHistoryText: { color: '#AAA', fontSize: moderateScale(13), marginTop: 10 },
  sortStatus: { flexDirection: 'row', alignItems: 'center', marginBottom: 10 },
  sortText: { fontSize: 12, color: '#666', marginLeft: 5 },
});

export default styles;
